src/compiler/commandLineParser.ts

```diff
@@ -584,4 +581,768 @@ namespace ts {
+        }
+
+        return false;
+    }
+
+    /**
+     * Reads the contents of a directory for a specific kind of item. Results are
+     * cached for performance.
+     *
+     * @param basePath The path to the directory. The path must already exist.
+     * @param kind The kind of file system entry to find.
+     * @param context The expansion context.
+     */
+    function readDirectory(basePath: Path, kind: FileSystemEntryKind, context: ExpansionContext) {
+        const { cache, host } = context;
+
+        let entry = cache.get(basePath);
+        if (entry === undefined) {
+            entry = {};
+            cache.set(basePath, entry);
+        }
+
+        if (typeof entry === "object") {
+            if (kind === FileSystemEntryKind.File) {
+                if (entry.files === undefined) {
+                    entry.files = host.readFileNames(basePath);
+                }
+
+                return entry.files;
+            }
+            else if (kind === FileSystemEntryKind.Directory) {
+                if (entry.directories === undefined) {
+                    entry.directories = host.readDirectoryNames(basePath);
+                }
+
+                return entry.directories;
+            }
+        }
+
+        return [];
+    }
+
+    /**
+     * Determines whether a literal or wildcard file has already been included that has a higher
+     * extension priority.
+     *
+     * @param file The path to the file.
+     * @param extensionPriority The priority of the extension.
+     * @param context The expansion context.
+     */
+    function hasFileWithHigherPriorityExtension(file: Path, extensionPriority: ExtensionPriority, context: ExpansionContext) {
+        const { literalFiles, wildcardFiles, supportedExtensions } = context;
+        const adjustedExtensionPriority = adjustExtensionPriority(extensionPriority);
+        for (let i = ExtensionPriority.Highest; i < adjustedExtensionPriority; ++i) {
+            const higherPriorityExtension = supportedExtensions[i];
+            const higherPriorityPath = changeExtension(file, higherPriorityExtension);
+            if (literalFiles.contains(higherPriorityPath) || wildcardFiles.contains(higherPriorityPath)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Removes files included via wildcard expansion with a lower extension priority that have
+     * already been included.
+     *
+     * @param file The path to the file.
+     * @param extensionPriority The priority of the extension.
+     * @param context The expansion context.
+     */
+    function removeWildcardFilesWithLowerPriorityExtension(file: Path, extensionPriority: ExtensionPriority, context: ExpansionContext) {
+        const { wildcardFiles, supportedExtensions } = context;
+        const nextExtensionPriority = getNextLowestExtensionPriority(extensionPriority);
+        for (let i = nextExtensionPriority; i < supportedExtensions.length; ++i) {

```

Comment:
No it cannot, at least not directly. I am starting from an explicit priority, not zero.